common/models.py

```diff
@@ -0,0 +1,107 @@
+import logging
+
+from django.db import models
+from django.utils import timezone
+from django.conf import settings
+
+LOGGER = logging.getLogger(__file__)
+
+CONTACT_TYPES = (
+    ('EMAIL', 'email'),
+    ('LANDLINE', 'landline'),
+    ('MOBILE', 'Mobile Phone number'),
+
+)
+
+
+class AbstractBase(models.Model):
+    """
+    Provides auditing attributes to a model.
+    It will provide audit fields that will keep track of when a model
+    is created or updated and by who.
+    """
+
+    created = models.DateTimeField(default=timezone.now)
+    updated = models.DateTimeField(default=timezone.now)
+    created_by = models.ForeignKey(
+        settings.AUTH_USER_MODEL, related_name='+')
+    updated_by = models.ForeignKey(
+        settings.AUTH_USER_MODEL, related_name='+')
+
+    def preserve_created_and_created_by(self):
+        """
+        Ensures that in subsequent times created and created_by fields
+        values are not overriden.
+        """
+
+        try:
+            original = self.__class__.objects.get(pk=self.pk)
+            self.created = original.created
+            self.created_by = original.created_by
+        except self.__class__.DoesNotExist:
+            LOGGER.info(
+                'preserve_created_and_created_by '
+                'Could not find an instance of {} with pk {} hence treating '
+                'this as a new record.'.format(self.__class__, self.pk))
+
+    def save(self, *args, **kwargs):
+        self.preserve_created_and_created_by()
+        super(AbstractBase, self).save(*args, **kwargs)
+
+    class Meta:
+        abstract = True
+
+
+class RegionAbstractBase(AbstractBase):
+    """
+    Model to hold the common attributes of a region
```

Comment:
This docstring does not tell us what a "region" is.

Remember - a docstring's job is not to explain things that are already manifest in the code, but to explain the _why_ and expose any potential traps / pitfalls.